metric-example/query-by-example.md

```diff
@@ -0,0 +1,516 @@
+
+Some characters have a special meaning in metric selectors: ( ) : , ~ "
+
+Additionally, white space is used to tell where an identifier such as "dt.entity.host" ends. Replacing the periods (full stops) with spaces would result in three identifiers next to each other, rather than a single one that contains white space.
+
+If white space or any of mentioned characters are to be used within a dimension name, in a filter matcher reference, or otherwise in their "literal" sense, rather than as syntax, we need to be careful and escape or quote them. Knowing about proper escaping is especially important when we build our selectors in code that handles user input. In such scenarios, e.g. when a user can provide a filter reference in a text field, we want to prevent them from making the resulting selector syntactically invalid or even worse, allow them to change its meaning and insert a malicious piece of code into our selector.
+
+Say, our user legitimately wants to filter for a host with name `Host 14.A "Alice" ~80° (Europe, North Africa)`. This filter reference can not be inserted into the selector as it is, since it contains a period, double quotes, a tilde, brackets, spaces and a comma. In order not to understand these characters and syntax, we surround the string with double quotes, and escape the contained double quotes with a tilde to indicate that the quoted part is not over yet, but that we have an instance of a literal quote:
+```
+"Host 14.A ~"Alice~" ~~80° (Europe, North Africa)"
+```
+Alternatively, we can rely entirely on escaping instead of quoting, but this gets messy real quick:
+```
+Host~ 14~.A~ "Alice"~ ~~80° ~(Europe~,~ North~ Africa~)
+```
+To filter for this name, we can use a combination of :names and :filter and use our quoted version for the filter matcher to end up with a valid selector that uses special characters:
+```
+builtin:host.disk.avail  
+: names  
+: filter(  
+	eq(  
+		dt.entity.host.name,  
+		"Host 14.A ~"Alice~" ~~80° (Europe, North Africa)"   
+	)  
+ )
+```
+
+# Scenario 16: Two Metrics with Two Different Entity Selectors
+
+>**Task**
+>We want to query multiple metrics at once, and each metric should use a different entity selector.
+
+There is only one `entitySelector` GET parameter, and an entity selector queries for exactly one entity type, but the metric selector can query more than one metric and those metrics could use different entity types.
+
+How can we query the PGI-based metric `builtin:tech.generic.cpu.usage` together with `builtin:host.cpu.usage` with the hosts and process group instances both filtered using a tag "business-critical"?
+
+This is easily possible by embedding entity selectors into filters using the `in` matcher with the right-hand side of `in` supplied using the `entitySelector` function that sets off an entity query from within a metric selector. Since entity selectors use a lot of special characters, the argument to the `entitySelector` function should always be quoted:

```

Comment:
```suggestion
This is easily possible by embedding entity selectors into filters using the `in` matcher. The right-hand side of `in` can be supplied with an `entitySelector` function that runs an entity query from within a metric selector. Since entity selectors use a lot of special characters, the argument of the `entitySelector` function should always be quoted:

```